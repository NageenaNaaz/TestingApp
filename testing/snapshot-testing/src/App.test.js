import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import PositiveCounter from './PositiveCounter';

jest.mock('react-dom');

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('should render 0', () => {
  const tree = renderer.create(<PositiveCounter/>).toJSON();
  expect(tree).toMatchSnapshot();
});


it('should render 2', () => {
  const component = renderer.create(<PositiveCounter/>);
  component.getInstance().increment();
  component.getInstance().increment();
  expect(component.toJSON()).toMatchSnapshot();
});


it('should not go negative', () => {
  const component = renderer.create(<PositiveCounter/>);
  component.getInstance().increment();
  component.getInstance().decrement();
  component.getInstance().decrement();
  expect(component.toJSON()).toMatchSnapshot();
});
it('renders a snapshot', () => {
  const tree = renderer.create(<App/>).toJSON();
  expect(tree).toMatchSnapshot();
});
