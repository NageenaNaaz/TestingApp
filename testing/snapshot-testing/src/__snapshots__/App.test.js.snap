// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h2>
      Welcome to React
    </h2>
  </div>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
  <span>
    Value: 
    0
    <button
      className="decrement"
      onClick={[Function]}
    >
      −
    </button>
    <button
      className="increment"
      onClick={[Function]}
    >
      +
    </button>
  </span>
</div>
`;

exports[`should not go negative 1`] = `
<span>
  Value: 
  0
  <button
    className="decrement"
    onClick={[Function]}
  >
    −
  </button>
  <button
    className="increment"
    onClick={[Function]}
  >
    +
  </button>
</span>
`;

exports[`should render 0 1`] = `
<span>
  Value: 
  0
  <button
    className="decrement"
    onClick={[Function]}
  >
    −
  </button>
  <button
    className="increment"
    onClick={[Function]}
  >
    +
  </button>
</span>
`;

exports[`should render 2 1`] = `
<span>
  Value: 
  2
  <button
    className="decrement"
    onClick={[Function]}
  >
    −
  </button>
  <button
    className="increment"
    onClick={[Function]}
  >
    +
  </button>
</span>
`;
